stores/_test-utils/src/domains/memory/resources.ts

```diff
@@ -10,7 +10,16 @@ export function createResourcesTest({ storage }: { storage: MastraStorage }) {
 
       // Save resource
       const savedResource = await storage.saveResource({ resource });
-      expect(savedResource).toEqual(resource);
+      expect(savedResource.id).toEqual(resource.id);
+      expect(savedResource.workingMemory).toEqual(resource.workingMemory);
+      expect(savedResource.metadata).toEqual(resource.metadata);
+      // Compare timestamps within 5ms tolerance to avoid flakiness from precision differences
+      expect(Math.abs(savedResource.updatedAt.getTime() - resource.updatedAt.getTime())).toBeLessThan(5);
+      expect(Math.abs(savedResource.createdAt.getTime() - resource.createdAt.getTime())).toBeLessThan(5);
```

Comment:
probably need to check that it's greater than -5 too

Reply:
@DanielSLew I think if we use mock/fake dates then we could possibly miss real timestamp precision issues introduced at some point in the future.  So far I've only seen dynamodb be off by 1 microsecond and everything else is spot on.